src/editor.ts

```diff
@@ -1,265 +1,272 @@
-        .composite as keyof languagePreferenceType;
-    }
-    return urls[mode][preference];
-  }
-
-  async _render(): Promise<void> {
-    // now set the src accordingly on the iframe....?
-    const filePath = this._context.path;
-    const sessionId = this._context.model.session;
-    // dig out the user from the URL; it will be the component of the path with an @ in it
-    const parentUrl = window.location.href;
-    const components = parentUrl.split('/');
-    const userComponents = components.filter(comp => comp.indexOf('@') >= 0);
-    const user = userComponents.length > 0 ? userComponents[0] : '';
-    // assemble the url and load it into the iframe
-    // const baseURL =
-    //   'https://matt.engagelively.com/users/rick/published/Dashboard%20Studio%20Development/index.html?';
-    // for debug:
-    // const baseURL =
-    //   'https://matt.engagelively.com/worlds/load?name=Dashboard%20Studio%20Development&';
-    // both of these are defined in constants.ts, NOT versioned to avoid bogus changes and stashes.
-    const baseURL = await this._baseUrl();
-    this._iframe.src = `${baseURL}dashboard_file=${filePath}&session=${sessionId}&inJupyterLab=true&user=${user}`;
-    // wait for session to load
-  }
-
-  setOption(key: string, value: any): void {
-    // do nothingß
-  }
-
-  setOptions(options: Partial<CodeEditor.IConfig>): void {
-    // do nothingß
-  }
-
-  getCursorPosition(): CodeEditor.IPosition {
-    return {
-      line: 0,
-      column: 0
-    };
-  }
-
-  getLine(lineNumber: number): string | null {
-    return null;
-  }
-
-  get host(): HTMLElement {
-    return this._iframe;
-  }
-}
-
-/**
- * A namespace for TableOfContents statics.
- */
-export namespace GalyleoEditor {
-  /**
-   * Interface describing table of contents widget options.
-   */
-  export interface IOptions {
-    /**
-     * Application document manager.
-     */
-    // docmanager: IDocumentManager;
-    context: DocumentRegistry.IContext<GalyleoModel>;
-    settings: ISettingRegistry;
-
-    /**
-     * Notebook ref.
-     */
-    // notebook: INotebookTracker;
-
-    // labShell: ILabShell;
-
-    // app: JupyterFrontEnd;
-
-    // browserModel: FileBrowserModel;
-  }
-
-  /**
-   * Interface describing the current widget.
-   */
-  export interface ICurrentWidget<W extends Widget = Widget> {
-    /**
-     * Current widget.
-     */
-    widget: W;
-  }
-}
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { Widget } from '@lumino/widgets';
+import {
+  DocumentRegistry,
+  DocumentWidget,
+  DocumentModel
+} from '@jupyterlab/docregistry';
+import { PLUGIN_ID } from './index';
+// import { baseURL } from './constants';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { CodeEditor } from '@jupyterlab/codeeditor';
+
+export class GalyleoDocument extends DocumentWidget<
+  GalyleoEditor,
+  DocumentModel
+> {}

```

Comment:
You could also simply define a type.

```suggestion
export type GalyleoDocument = DocumentWidget<GalyleoEditor, DocumentModel>;

```